Source/WebGPU/WebGPU/BindGroupLayout.mm

```diff
@@ -141,13 +140,36 @@
     if (storageTexture.nextInChain)
         return nil;
 
-    UNUSED_PARAM(storageTexture);
     auto descriptor = [MTLArgumentDescriptor new];
     descriptor.dataType = MTLDataTypeTexture;
     // FIXME: Implement this.
     return descriptor;
 }
 
+bool BindGroupLayout::isPresent(const WGPUExternalTextureBindingLayout&)
+{
+    return true;
+}
+
+static MTLArgumentDescriptor *createArgumentDescriptor(const WGPUExternalTextureBindingLayout& externalTexture)
```

Comment:
I don't think this function is necessary. It duplicates `static MTLArgumentDescriptor *createArgumentDescriptor(const WGPUTextureBindingLayout& texture)`

Reply:
We can certainly leave this for a second patch, but this could be implemented by returning an array of MTLArgumentDescriptor instances and using `append` instead of `unchckedAppend` when adding the descriptors

Reply:
Right, I created it for symmetry; there is a `createArgumentDescriptor()` for each `WGPU*BindingLayout` type. The fact that it matches right now is temporary; as the comment describes, the content of this function is totally bogus right now and will need to be fully implemented in a follow-up patch. So I think the function should still exist; it's just an unlucky accident that it does the same thing as the `WGPUTextureBindingLayout` version right now.

You're right that returning a single `MTLArgumentDescriptor` ain't gonna work when we implement this properly.